File: js/packages/app/src/__tests__/features/global-functions/wrappers/global-function-example/src/polywrap/external/module/WrapModule.ts
import { IWrapInstance } from "@polywrap/reim-wrap-as";
import { ImportBindings } from "./ImportBindings";




export class WrapModule {
  static wrapInstance: IWrapInstance = new ExternalWrapInstance();

  static connect(instance: IWrapInstance): void {
    WrapModule.wrapInstance = instance;
  }

  static import(instance: IWrapInstance): ImportBindings {
    return new ImportBindings(

                                    
    );
  }
}

File: rust/test_cases/wrappers/global_functions/src/polywrap/external/module/wrap_module.rs
use std::sync::Arc;

use reim_dt::ExternalModule;
use crate::polywrap::external::module::import_bindings::{ ImportBindings };

use crate::polywrap::external::global_functions::test_host_global_function::{ create as createTestHostGlobalFunction };

use crate::polywrap::external::module::external_wrap_module::set_external_module;

pub struct WrapModule {}

impl WrapModule {
  pub async fn connect(external_module: Arc<dyn ExternalModule>) {
    set_external_module(external_module).await;
  }

  pub fn import(instance: Arc<dyn ExternalModule>) -> ImportBindings {
    ImportBindings::new(
                                                      
      createTestHostGlobalFunction(Arc::clone(&instance)),
    )
  }
}
